23520 - Handle Continuation Draft States

ArwenQin opened the pull request:
*Issue #:* /bcgov/entity#23520

*Description of changes:*
- Handle CHANGE_REQUESTED status: reason in gray block; change wording; show NR expiry
- Handle APPROVED status: show NR expiry, authorization expiry, split button to delete
- Handle NR approved draft status: change button label, show NR expiry
- RESUBMITTED & AWAITING_REVIEW are handled already, updated wording
- REJECTED & APPROVED application are handled already


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

ArwenQin commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://business-filings-dev--pr-696-pgczq89k.web.app

ArwenQin commented on src/components/Dashboard/TodoList.vue:
```diff
@@ -603,15 +648,47 @@
+                NR APPROVED - {{ expiresText(getNameRequest) }}
+              </p>
+              <p class="list-item__subtitle">
+                Authorization APPROVED - Expires in {} days
```
authorization expiry date property is not implemented yet?

severinbeauvais replied:
Oh, this is obsolete. We removed this date. Can you update, please?

ArwenQin replied:
Removed

ArwenQin replied:
![image](https://github.com/user-attachments/assets/b0d4702c-ef34-4fde-b701-03ae2ec18749)

ArwenQin commented:
CHANGE_REQUESTED 
try: https://business-filings-dev--pr-696-pgczq89k.web.app/Tl4Z79Y9h4
![image](https://github.com/user-attachments/assets/bb3ac99c-73ad-4790-add6-13a56020b0cc)

ArwenQin commented:
APPROVED can't demo, fake the status locally:
![image](https://github.com/user-attachments/assets/f50b9882-3a3f-4cae-b047-38cf66fd4c7f)

ArwenQin commented:
NR Approved draft can't demo due to NR system issue

ArwenQin commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://business-filings-dev--pr-696-pgczq89k.web.app

ArwenQin commented:
AWAITING_REVIEW:
try: https://business-filings-dev--pr-696-pgczq89k.web.app/TqPo3HWiXN
![image](https://github.com/user-attachments/assets/f4dbf9ef-853c-4f20-be5f-cea83473bf1c)
Updated some wording to match design

ArwenQin commented:
REJECTED & APPROVED are handled already, no change from now
https://business-filings-dev--pr-696-pgczq89k.web.app/T0uCJs8voi
![image](https://github.com/user-attachments/assets/972e1cb7-cf33-43bb-9c2a-81eb027a2c58)

https://business-filings-dev--pr-696-pgczq89k.web.app/TMVgAUtu61
![image](https://github.com/user-attachments/assets/15c3ccea-2102-46ae-8e19-5cbb0fca0d8b)
![image](https://github.com/user-attachments/assets/c48f01b7-d652-4555-b602-e36b8d3ef71c)

severinbeauvais commented:
> APPROVED can't demo, fake the status locally: ![image](https://private-user-images.githubusercontent.com/122495122/375502559-f50b9882-3a3f-4cae-b047-38cf66fd4c7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1OTA0NTIsIm5iZiI6MTcyODU5MDE1MiwicGF0aCI6Ii8xMjI0OTUxMjIvMzc1NTAyNTU5LWY1MGI5ODgyLTNhM2YtNGNhZS1iMDQ3LTM4Y2Y2NmZkNGM3Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxMFQxOTU1NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02YThiNWNlNmY3YmM5NzYxNDdhZjYwNjM4N2IwZTAyNjE2ZmE4YzY5OTM2OGJmOTRiNDU1OGIwNjBmMGVjYTg2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5SeaBNgZW_dcp3eaDjJqlb9_eXt4WiM7-72KKNV9dDU)

This is missing a subtitle (in the large white space between the title and content).

severinbeauvais commented:
> REJECTED & APPROVED (whole application is approved) are handled already, no change from now https://business-filings-dev--pr-696-pgczq89k.web.app/T0uCJs8voi ![image](https://private-user-images.githubusercontent.com/122495122/375509884-972e1cb7-cf33-43bb-9c2a-81eb027a2c58.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1OTA0NTIsIm5iZiI6MTcyODU5MDE1MiwicGF0aCI6Ii8xMjI0OTUxMjIvMzc1NTA5ODg0LTk3MmUxY2I3LWNmMzMtNDNiYi05YzJhLTgxZWIwMjdhMmM1OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxMFQxOTU1NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MGE0ZDNiYzY2MzRmZWRlZWRlMWQyODc5MTA4YjNkOTU4Yjc5YTMyOGFmYmRkZGE5NmRlZGE3ODRmZGFiNTkzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.oEZadHlirZA53kHuH35VXTCu6NrbFU1knLikC4uB8uI)

Can you please style the staff comment same as Change Requested?

ArwenQin commented:
> > APPROVED can't demo, fake the status locally: ![image](https://private-user-images.githubusercontent.com/122495122/375502559-f50b9882-3a3f-4cae-b047-38cf66fd4c7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1OTA0NTIsIm5iZiI6MTcyODU5MDE1MiwicGF0aCI6Ii8xMjI0OTUxMjIvMzc1NTAyNTU5LWY1MGI5ODgyLTNhM2YtNGNhZS1iMDQ3LTM4Y2Y2NmZkNGM3Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxMFQxOTU1NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02YThiNWNlNmY3YmM5NzYxNDdhZjYwNjM4N2IwZTAyNjE2ZmE4YzY5OTM2OGJmOTRiNDU1OGIwNjBmMGVjYTg2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5SeaBNgZW_dcp3eaDjJqlb9_eXt4WiM7-72KKNV9dDU)
> 
> This is missing a subtitle (in the large white space between the title and content).

May I know where to see the design with the subtitle? I only find this:
![image](https://github.com/user-attachments/assets/f391f5f4-d38b-427c-83cb-ae06e94b793e)
I will remove the authorization expiry date too.

severinbeauvais commented on src/components/Dashboard/TodoList.vue:
```diff
@@ -521,6 +520,52 @@
+                          mdi-delete-forever
+                        </v-icon>
+                        <v-list-item-title>
+                          Delete {{ EnumUtilities.filingTypeToName(item.name) }}
```
No. Once the authorization is submitted (ie, a review record is created), the user can no longer delete the filing.

The user can only delete the filing when they're at the authorization stage (including save and resume later) and before they submit it for review.

ArwenQin replied:
Removed the drop down menu

severinbeauvais commented on src/components/Dashboard/TodoList.vue:
```diff
@@ -603,15 +648,47 @@
             <PaymentPending v-else />
           </template>
 
+          <!-- is this an NR approved continuation draft item? -->
```
Let's call this a "continuation authorization item".

ArwenQin replied:
Line 651 - 661 are not needed, I will remove them. This is already handled in line 1613

severinbeauvais replied:
OK, but if you see an opportunity to update the comments, please call this filing in Draft, Pending, Change Requested statuses a "continuation authorization", while the later statuses are a "continuation application". This will be consistent with what they're called in Create UI and elsewhere.

severinbeauvais commented:
> APPROVED can't demo, fake the status locally: ![image](https://private-user-images.githubusercontent.com/122495122/375502559-f50b9882-3a3f-4cae-b047-38cf66fd4c7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1OTEwODMsIm5iZiI6MTcyODU5MDc4MywicGF0aCI6Ii8xMjI0OTUxMjIvMzc1NTAyNTU5LWY1MGI5ODgyLTNhM2YtNGNhZS1iMDQ3LTM4Y2Y2NmZkNGM3Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxMFQyMDA2MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYzFiZmZmMTRmYmUyN2NmOTIwNjZlOTUwNjIwNDJmNjI2YTVhMDcwYWY3Nzk1YjU2NGZiMWM2MDNmNDZhMmE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.9DNZGm6tSqapXH0MQjEJkcVDojPoJZffBs5ATOGrdXs)

Here's a sample Draft continuation (ie, authorization): https://dev.business.bcregistry.gov.bc.ca/TOWa0a3U5H

severinbeauvais commented on src/components/Dashboard/TodoList.vue:
```diff
@@ -2442,6 +2523,8 @@ export default class TodoList extends Mixins(AllowableActionsMixin, DateMixin) {
 textarea {
   color: $gray7;
+  background-color: $gray1;
+  padding: 10px 10px 10px 20px;
```
If it works, I recommend moving this padding to the textarea component in the HTML and using the Vuetify helper classes.

By the way, I think should specify 20px right padding, too. (See px and py classes.)

ArwenQin replied:
Updated

severinbeauvais commented:
> > > APPROVED can't demo, fake the status locally: 
> > 
> > This is missing a subtitle (in the large white space between the title and content).
> 
> May I know where to see the design with the subtitle?

Oh, you have the latest design. I believe both approved statements are part of the subtitle? Anyway, they should display while the item is collapsed and not have so much whitespace in there.

If you show the details, you should see the NR info, I think.

ArwenQin commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://business-filings-dev--pr-696-pgczq89k.web.app

ArwenQin commented:
Updated for the APPROVED status:
![image](https://github.com/user-attachments/assets/3df016a6-4c2b-4c4e-a4d8-22e64b4aa2d4)

![image](https://github.com/user-attachments/assets/5ffdcba4-ef27-4305-9959-f0b694a19806)

Removed the blank subtitle; expand can see the NR details; removed the drop down menu to delete; remove authorization expiry

ArwenQin commented:
Update for the REJECTED status:
![image](https://github.com/user-attachments/assets/9335dc0a-b44d-4988-9b33-bf847d672ecc)

style the comments section

severinbeauvais commented:
> Updated for the APPROVED status: ![image](https://private-user-images.githubusercontent.com/122495122/375559342-3df016a6-4c2b-4c4e-a4d8-22e64b4aa2d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2MDA0ODUsIm5iZiI6MTcyODYwMDE4NSwicGF0aCI6Ii8xMjI0OTUxMjIvMzc1NTU5MzQyLTNkZjAxNmE2LTRjMmItNGM0ZS1hNGQ4LTIyZTY0YjRhYTJkNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxMFQyMjQzMDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYzllY2YwZjY5ZDBlNmY2YjgyYTU0Mjc2ZmUwZDM0ZGU1YjI1OTAyYjMzOTlmMGFiNGU1MmI1MjFhMmE1OTg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.l2z0u7fukQVY_XTq9cPC9XVJRyfwz06keOVPx5fh18g)

I think there needs to be an extra sub-title to show the staff approved the authorization ...?

severinbeauvais commented on src/components/Dashboard/TodoList.vue:
```diff
@@ -603,18 +627,29 @@
+              </p>
               <p class="list-item__subtitle">
-                This {{ item.title }} is paid but requires you to make the following changes:
+                BC Registries staff require the following changes to your authorization documentation
```
Ask Jacqueline if there should be a `:` at the end of the line.

ArwenQin replied:
checking now

ArwenQin commented:
> > Updated for the APPROVED status: ![image](https://private-user-images.githubusercontent.com/122495122/375559342-3df016a6-4c2b-4c4e-a4d8-22e64b4aa2d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2MDA0ODUsIm5iZiI6MTcyODYwMDE4NSwicGF0aCI6Ii8xMjI0OTUxMjIvMzc1NTU5MzQyLTNkZjAxNmE2LTRjMmItNGM0ZS1hNGQ4LTIyZTY0YjRhYTJkNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxMFQyMjQzMDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYzllY2YwZjY5ZDBlNmY2YjgyYTU0Mjc2ZmUwZDM0ZGU1YjI1OTAyYjMzOTlmMGFiNGU1MmI1MjFhMmE1OTg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.l2z0u7fukQVY_XTq9cPC9XVJRyfwz06keOVPx5fh18g)
> 
> I think there needs to be an extra sub-title to show the staff approved the authorization ...?

Maybe keep the "Authorization APPROVED"?
![image](https://github.com/user-attachments/assets/0ddccc8c-799e-4a26-8285-988f9fab54a0)

severinbeauvais commented:
> Maybe keep the "Authorization APPROVED"? ![image](https://private-user-images.githubusercontent.com/122495122/375567565-0ddccc8c-799e-4a26-8285-988f9fab54a0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2MDA5ODcsIm5iZiI6MTcyODYwMDY4NywicGF0aCI6Ii8xMjI0OTUxMjIvMzc1NTY3NTY1LTBkZGNjYzhjLTc5OWUtNGEyNi04Mjg1LTk4OGY5ZmFiNTRhMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxMFQyMjUxMjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03Nzc5NWY1MjhjNTQyMTNkODFmM2Y4YjVjZjdiOTgwN2U2N2RlOWYxZmZjMDA2NjJkNzNhZjM4NDBkYmI0N2RiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.WftrkSuKSdojkeWJqN3xzog0_DyMPSkJVi4Qfc-QqLE)

Yes, I think so.
